language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/ExecutorPositionsUtil.java

```diff
@@ -0,0 +1,185 @@
+                                        == ProjectKind.SINGLE_FILE_PROJECT
+                                        && Files.isSameFile(filePath, project.sourceRoot())));
+                            } catch (IOException e) {
+                                return false;
+                            }
+                        })
+                        .map(symbol -> (ServiceDeclarationSymbol) symbol)
+                        .collect(Collectors.toList())
+                        .forEach(serviceSymbol -> {
+                            JsonObject serviceObject = new JsonObject();
+                            serviceObject.addProperty(KIND, SOURCE);
+                            if (serviceSymbol.getLocation().isPresent()) {
+                                Location location = serviceSymbol.getLocation().get();
+                                setRange(location, serviceObject, location.lineRange().filePath());
+                            }
+                            Optional<ServiceAttachPoint> serviceAttachPoint = serviceSymbol.attachPoint();
+                            if (serviceAttachPoint.isPresent()) {
+                                ServiceAttachPointKind kind = serviceAttachPoint.get().kind();
+                                if (kind == ServiceAttachPointKind.ABSOLUTE_RESOURCE_PATH) {
+                                    List<String> segments =
+                                            ((AbsResourcePathAttachPoint) serviceAttachPoint.get()).segments();
+                                    serviceObject.addProperty(NAME, String.join("_", segments));
+                                } else if (kind == ServiceAttachPointKind.STRING_LITERAL) {
+                                    serviceObject.addProperty(NAME,
+                                            ((LiteralAttachPoint) serviceAttachPoint.get()).literal());
+                                }
+                            }
+                            execPositions.add(serviceObject);
+                        });
+            }
+            getTestCasePositions(execPositions, project, filePath, module);
+        });
+        return execPositions;
+    }
+
+    protected static void setRange(Location location, JsonObject jsonObject, String filePath) {
+        JsonObject rangeObject = new JsonObject();
+        JsonObject startObject = new JsonObject();
+        JsonObject endObject = new JsonObject();
+        LineRange lineRange = location.lineRange();
+        startObject.addProperty(LINE, lineRange.startLine().line());
+        startObject.addProperty(CHARACTER, lineRange.startLine().offset());
+        endObject.addProperty(LINE, lineRange.endLine().line());
+        endObject.addProperty(CHARACTER, lineRange.endLine().offset());
+        jsonObject.addProperty(FILE_PATH, filePath);
+        rangeObject.add(START, startObject);
+        rangeObject.add(END, endObject);
+        jsonObject.add(RANGE, rangeObject);
```

Comment:
What if we create an POJO here and then convert it to json? is it possible?

Reply:
Updated with https://github.com/ballerina-platform/ballerina-lang/pull/30125/commits/460e141f5a8ee7388845f7fd6856c5117f898d82

Reply:
Updated with https://github.com/ballerina-platform/ballerina-lang/pull/30125/commits/460e141f5a8ee7388845f7fd6856c5117f898d82